packages/core/src/routes/admin-user.ts

```diff
@@ -122,10 +125,146 @@ export default function adminUserRoutes<T extends AuthedRouter>(router: T) {
     }
   );
 
+  router.patch(
+    '/users/:userId/password',
+    koaGuard({
+      params: object({ userId: string() }),
+      body: object({ password: string().min(6) }),
+    }),
+    async (ctx, next) => {
+      const {
+        params: { userId },
+        body: { password },
+      } = ctx.guard;
+
+      await findUserById(userId);
+
+      const { passwordEncrypted, passwordEncryptionSalt } = encryptUserPassword(userId, password);
+
+      const user = await updateUserById(userId, {
+        passwordEncryptionSalt,
+        passwordEncrypted,
+      });
+      ctx.body = pick(user, ...userInfoSelectFields);
+
+      return next();
+    }
+  );
+
+  router.patch(
+    '/users/:userId/username',
+    koaGuard({
+      params: object({ userId: string() }),
+      body: object({ username: string().min(3) }),
+    }),
+    async (ctx, next) => {
+      const {
+        params: { userId },
+        body: { username },
+      } = ctx.guard;
+
+      assertThat(
+        !(await hasUser(username)),
+        new RequestError({
+          code: 'user.username_exists_register',
+          status: 422,
+        })
+      );
+
+      const user = await updateUserById(userId, {
+        username,
+      });
+
+      ctx.body = pick(user, ...userInfoSelectFields);
+
+      return next();
+    }
+  );
+
+  router.patch(
+    '/users/:userId/primary-email',
+    koaGuard({
+      params: object({ userId: string() }),
+      body: object({ primaryEmail: string().email() }),
```

Comment:
i remember that 'string().email()' still leads to 'string()' type check. Maybe use emailRegEx instead (in this case, need to refactor the check logic a little)?
<img width="434" alt="image" src="https://user-images.githubusercontent.com/15182327/154645505-72b721dd-3e7f-4f6e-8b60-7fee8eaa4e29.png">

Reply:
good call. updated